src/main/java/io/kestra/plugin/aws/runner/AwsBatchScriptRunner.java

```diff
@@ -0,0 +1,546 @@
+        BatchClientBuilder batchClientBuilder = BatchClient.builder()
+            .credentialsProvider(AbstractConnectionInterface.credentialsProvider(this.awsClientConfig(runContext)))
+            .region(Region.of(renderedRegion))
+            // Use the httpClientBuilder to delegate the lifecycle management of the HTTP client to the AWS SDK
+            .httpClientBuilder(serviceDefaults -> ApacheHttpClient.builder().build());
+
+        if (this.endpointOverride != null) {
+            batchClientBuilder.endpointOverride(URI.create(runContext.render(this.endpointOverride)));
+        }
+
+        BatchClient client = batchClientBuilder
+            .build();
+
+        String jobName = IdUtils.create();
+        ContainerProperties.Builder containerPropsBuilder = ContainerProperties.builder()
+            .image(runContext.render(commands.getContainerImage()))
+            .logConfiguration(
+                LogConfiguration.builder()
+                    .logDriver(LogDriver.AWSLOGS)
+                    .options(Map.of("awslogs-stream-prefix", jobName))
+                    .build()
+            );
+
+        if (this.resources != null && this.resources.getRequest() != null) {
+            containerPropsBuilder
+                .resourceRequirements(
+                    ResourceRequirement.builder()
+                        .type(ResourceType.MEMORY)
+                        .value(this.resources.getRequest().getMemory())
+                        .build(),
+                    ResourceRequirement.builder()
+                        .type(ResourceType.VCPU)
+                        .value(this.resources.getRequest().getCpu())
+                        .build()
+                );
+        }
+
+        if (this.executionRoleArn != null) {
+            containerPropsBuilder.executionRoleArn(runContext.render(this.executionRoleArn));
+        }
+
+        if (commands.getEnv() != null) {
+            containerPropsBuilder
+                .environment(
+                    commands.getEnv().entrySet().stream()
+                        .map(e -> KeyValuePair.builder().name(e.getKey()).value(e.getValue()).build())
+                        .toArray(KeyValuePair[]::new)
+                );
+        }
+
+        String renderedBucket = runContext.render(s3Bucket);
+        String workingDirKey = IdUtils.create();
+        String outputDirKey = IdUtils.create();
+        Map<String, Object> additionalVars = Optional.ofNullable(renderedBucket)
+            .map(bucket -> Map.<String, Object>of(
+                "workingDir", "s3://" + bucket + "/" + workingDirKey,
+                "outputDir", "s3://" + bucket + "/" + outputDirKey
+            ))
+            .orElse(Collections.emptyMap());
+        List<String> command = ScriptService.uploadInputFiles(runContext, runContext.render(commands.getCommands(), additionalVars));
+
+        if (filesToUpload != null && !filesToUpload.isEmpty()) {
```

Comment:
```suggestion
        if (ListUtils.isEmpty(filesToUpload)) {
```
I added a small helper method about that in core ;)